src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -42,15 +53,11 @@ class Crystal::Doc::Generator
       filename = "Readme.md"
     end
 
-    if filename
-      body = File.read(filename)
-    else
-      body = ""
-    end
-
-    body = String.build do |io|
-      Markdown.parse body, MarkdownDocRenderer.new(program_type, io)
-    end
+    body = if filename
```

Comment:
fixed. although I would have done...

```
body = filename ? doc(program_type, File.read(filename)) : ""
```

... but i don't think it would have been received well